packages/otelbin/src/components/share/DownloadButton.tsx

```diff
@@ -0,0 +1,67 @@
+// SPDX-FileCopyrightText: 2023 Dash0 Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import React from 'react';
+import { useReactFlow, getRectOfNodes, getTransformForBounds } from 'reactflow';
+import { ImageDown } from "lucide-react";
+import { Button } from "~/components/button";
+import { toPng } from 'html-to-image';
+
+const downloadImage = (dataUrl: string) => {
+  const a = document.createElement('a');
+
+  a.setAttribute('download', 'collector-image.png');
+  a.setAttribute('href', dataUrl);
+  a.click();
+};
+
+const calculateImageDimensions = (nodeCount: number) => {
+  const baseWidth = 1024;
+  const baseHeight = 768;
+  const widthPerNode = 100; 
+
+  const calculatedWidth = Math.max(baseWidth, widthPerNode * nodeCount);
+
+  return {
+    width: calculatedWidth,
+    height: baseHeight,
+  };
+}
+
+export function DownloadButton() {
```

Comment:
**Suggestion:** Rename to `DownloadImageButton` to avoid confusion with the download as YAML capability.

Reply:
MSTM